charms/kfp-api/src/charm.py

```diff
@@ -474,6 +501,14 @@ def _on_remove(self, _):
                 raise error
         self.unit.status = MaintenanceStatus("K8S resources removed")
 
+    def _on_update_status(self, _):
+        """Update status actions."""
+        self._check_leader()
+        try:
+            self._check_status()
+        except ErrorWithStatus as err:
+            self.model.unit.status = err.status
```

Comment:
This is not absolutely needed for current function, but I think we should include it because the intent for this `except` block is it should set status and then exit.  There doesn't happen to be anything after so it doesn't matter now, but if someone added more code below in a future patch they might not realise check_status could error and then still continue.  

```suggestion
            self.model.unit.status = err.status
            return
```

Reply:
Noted. Will change.